buildimage.py

```diff
@@ -141,11 +141,22 @@ def golang_init():
         cmd = "go get github.com/mohae/deepcopy"
         run_cmd(cmd)
 
-        cmd = "git clone https://github.com/syncthing/syncthing " + syncthing_dir
-        run_cmd(cmd)
 
     return
 
+def syncthing_init():
+
+    # Check if syncthig is already cloned and pull latest if present else clone
+    if os.path.isdir(syncthing_dir):
+        log.info("Pulling from syncthing:master...")
+
+        cmd = "cd " + syncthing_dir + " && git pull origin master"
```

Comment:
change to "&& git pull"
No need to add 'origin master' 
In case, if git clone checks out any other branch, git pull will take code from that branch only...